database/src/models.rs

```diff
@@ -68,24 +68,27 @@ impl User {
     /// Checks if a plaintext password is valid.
     pub fn valid_password<S>(&self, password: S) -> bool
     where
-        S: Into<String>,
+        S: AsRef<str>,
     {
-        bcrypt::verify(&self.encrypted_password, &password.into())
+        bcrypt::verify(password.as_ref(), &self.encrypted_password)
```

Comment:
Note: this fixes a bug - the order of arguments to `bcrypt::verify` was flipped from what it should've been.